receiver/mongodbreceiver/config.go

```diff
@@ -25,11 +25,11 @@ type Config struct {
 	configtls.ClientConfig         `mapstructure:"tls,omitempty"`
 	// MetricsBuilderConfig defines which metrics/attributes to enable for the scraper
 	metadata.MetricsBuilderConfig `mapstructure:",squash"`
-	Hosts                         []confignet.AddrConfig `mapstructure:"hosts"`
-	Username                      string                 `mapstructure:"username"`
-	Password                      configopaque.String    `mapstructure:"password"`
-	ReplicaSet                    string                 `mapstructure:"replica_set,omitempty"`
-	Timeout                       time.Duration          `mapstructure:"timeout"`
+	Hosts                         []confignet.TCPAddrConfig `mapstructure:"hosts"`
```

Comment:
I think we could migrate more smoothly by adding the field as deprecated, and logging a warning during receiver start if the value is not empty.

```go
	// Deprecated - option will be removed in v0.102.0
	Transport TransportType `mapstructure:"transport"`
```

Reply:
I unresolved the comment because this has not been added yet

Reply:
I think you should leave `confignet.TCPAddrConfig` and add the separate `Transport` field as I showed above. This way, we can have everything working exactly as it will and only need to remove the deprecated field later.

Reply:
^ bump